department-of-veterans-affairs/vets-website
# Initial version of MHV sec nav

## Summary
Initial version of the MHV Secondary Nav. This version will still need styling tweaks to get it to the final version, but provides a way for the tool teams to start integrating it.

Feature toggle name: `mhv_secondary_navigation_enabled`

**For integrating the secondary nav into an application:**

- Import the following in your component or container that will use the secondary nav: `import { MhvSecondaryNav } from '@department-of-veterans-affairs/mhv/exports';`
- Import the following in the app entry file for your app: `import '~/platform/mhv/secondary-nav/sass/mhv-sec-nav.scss';`
  - ⚠️ This SASS file cannot be imported in any component or container as the unit test framework does not support SASS files.
- Add the following element to add the secondary nav. This element should be added to the top of your application where it can take the entire width of the page: `<MhvSecondaryNav />`

## Related issue(s)
https://app.zenhub.com/workspaces/mhv-on-vagov-landing-page-62619a987d74510018ecc546/issues/gh/department-of-veterans-affairs/va.gov-team/80821

## Testing done
- Added unit tests
- To run locally
  1. Run `yarn watch --env --entry=mhv-landing-page`
  2. Run `yarn mock-api --responses src/applications/mhv-landing-page/mocks/api/index.js`
  3. Navigate to `http://localhost:3001/my-health/`. Note that if you are redirect to login, then run the following command in your browser console: `localStorage.setItem('hasSession', true);` to reset your session.

## Screenshots
|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |    ![localhost_3001_my-health_(iPhone 14 Pro Max) (3)](https://github.com/department-of-veterans-affairs/vets-website/assets/63804190/849fdea3-5879-42c9-b1fe-36b870874322)   |    ![localhost_3001_my-health_(iPhone 14 Pro Max) (3)](https://github.com/department-of-veterans-affairs/vets-website/assets/63804190/defb7f92-b8af-473d-9383-309252fbc707)   |
| Desktop |      ![localhost_3001_my-health_ (8)](https://github.com/department-of-veterans-affairs/vets-website/assets/63804190/916c341e-1e75-43dc-8459-82510a9ec566)  |    ![localhost_3001_my-health_ (10)](https://github.com/department-of-veterans-affairs/vets-website/assets/63804190/ded5d69f-f1fc-4a7b-bd1b-d7aad1fd0a18)   |

## What areas of the site does it impact?
- MHV Landing page

## Acceptance criteria
- Implemented secondary nav component such that tool teams can add it to their apps (SM, Appts, Meds)
- Added secondary nav to MHV landing page behind the appropriate feature flag

### Quality Assurance & Testing


### Error Handling


### Authentication


### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:


## Requested Feedback
